config/src/lib.rs

```diff
@@ -2009,6 +2023,80 @@ impl<'a> Provider for RemappingsProvider<'a> {
     }
 }
 
+/// Generate warnings for unknown sections
+struct WarningsProvicer<P> {
```

Comment:
let's move this to another sub-module

we should have done this with other providers already, but let's start with this one -.-